File: K1.py
import pandas
import matplotlib.pyplot as plt
import seaborn

df = pandas.read_csv("C:\\Users\\bluew\\OneDrive\\桌面\\HW\\dye.csv")
fig, axes = plt.subplots(2, 2)

# barplot for nclass
nclass = df["class"].value_counts().sort_index()
seaborn.barplot(x=nclass.index, y=nclass.values, ax=axes[0, 0])
axes[0, 0].set_title("barplot for nclass")
axes[0, 0].set_xlabel(None)

# boxplot for c1, c2 and c3
seaborn.boxplot(df[["c1", "c2", "c3"]], ax=axes[0, 1], orient="v")
axes[0, 1].set_title("boxplot for c1, c2 and c3")

# Histogram of dye$c1
seaborn.histplot(df["c1"], ax=axes[1, 0], binwidth=0.2)
axes[1, 0].set_title("Histogram of dye$c1")
axes[1, 0].set_xlabel("dye$c1")
axes[1, 0].set_ylabel("Frequency")

# boxplot c3 for each class
seaborn.boxplot(df, x="class", y="c3", ax=axes[1, 1])
axes[1, 1].set_title("boxplot c3 for each class")
axes[1, 1].set_xlabel(None)
axes[1, 1].set_ylabel(None)

plt.show()

File: K1.jl
using CSV
using DataFrames
using Plots, StatsPlots

df = CSV.read("C:\\Users\\bluew\\OneDrive\\桌面\\HW\\dye.csv", DataFrame)

# barplot for nclass
val_count = Dict{UInt8, UInt16}()
[val_count[cls]=0 for cls in unique(df.class)]
[val_count[cls]+=1 for cls in df.class]
barPlot4nClass = bar(val_count)

# boxplot for c1, c2 and c3
x = repeat([1, 2, 3], nrow(df))
y = Matrix(df[:, [:"c1", :"c2", :"c3"]])'
y = vcat(y...)
boxPlot4c1c2c3 = boxplot(x, y)

# Histogram of dye$c1
histogramOfDyeC1 = histogram(df.c1)

# boxplot c3 for each class
x = df.class
y = df.c3
boxPlotC3ForEachClass = boxplot(x, y)

plot(barPlot4nClass, boxPlot4c1c2c3, histogramOfDyeC1, boxPlotC3ForEachClass, 
     layout=grid(2, 2), 
     legend=false, 
     xlabel=["" "" "dye\$c1" ""], 
     ylabel=["" "" "Frequency" ""], 
     title=["barplot for nclass" "boxplot for c1, c2 and c3" "Histogram of dye\$c1" "boxplot c3 for each class"]
)